### ISSUE TITLE
No way to avoid logging parsing errors

### ISSUE THREAD

**Issue by gigaroby** [mozilla-services/heka#1680]
cc: @vad
I am currently using ```DockerInput``` to parse some nginx log files on a machine where other containers are also running.
Heka tries to parse the output of all the containers and fails (rightfully) on those that are not nginx and logs the failures it in its log file.
By using a message matcher on the decoder I would be able to disallow the wrong containers by looking at ```Fields[ContainerName]``` but, unless I am mistaken there is no way to do so.

Since the errors are expected and they are just polluting the log file, is there a way to avoid it?

**Comment by andremedeiros** [comment]
Actually, you can do message matching with Fields, as described [here](http://hekad.readthedocs.org/en/v0.9.2/message_matcher.html#examples)

**Comment by gigaroby** [comment]
Yes, but I would need to match directly after the Input, after the message is dispatched to a Decoder and, as far as I know, there is no way to do that...

**Comment by trink** [comment]
It is not possible to use a message_matcher for a decoder.  In that case
you can create a sandbox decoder and do the log type identification/parsing
there for everything that comes out of your containers.

Trink

**Comment by trink** [comment]
See also #1398

**Comment by sathieu** [comment]
@gigaroby Try #1742 with print_decode_failures=false in your decoder config (see also #1743).
